Core/APIRequest.swift

```diff
@@ -34,43 +33,73 @@ public class APIRequest {
         
     }
     
+    public enum HTTPMethod: String {
+        case get = "GET"
+        case head = "HEAD"
+        case post = "POST"
+        case put = "PUT"
+        case delete = "DELETE"
+        case connect = "CONNECT"
+        case options = "OPTIONS"
+        case trace = "TRACE"
+        case patch = "PATCH"
+    }
+    
     @discardableResult
     public static func request(url: URL,
                                method: HTTPMethod = .get,
-                               parameters: [String: Any]? = nil,
-                               completion: @escaping APIRequestCompletion) -> Request {
+                               parameters: [String: String]? = nil,
+                               completion: @escaping APIRequestCompletion) -> URLSessionDataTask {
         os_log("Requesting %s", log: generalLog, type: .debug, url.absoluteString)
+                
+        let session = URLSession.shared
         
-        return Alamofire.request(url, method: method, parameters: parameters, headers: APIHeaders().defaultHeaders)
-            .validate(statusCode: 200..<300)
-            .responseData(queue: callbackQueue) { response in
+        let url = url.addParams(parameters ?? [:])
+        var urlRequest = URLRequest(url: url)
+        urlRequest.allHTTPHeaderFields = APIHeaders().defaultHeaders
+        urlRequest.httpMethod = method.rawValue
 
-                os_log("Request for %s completed with response code: %s and headers %s",
-                       log: generalLog,
-                       type: .debug,
-                       url.absoluteString,
-                       String(describing: response.response?.statusCode),
-                       String(describing: response.response?.allHeaderFields))
+        let task = session.dataTask(with: urlRequest) {
+            (data, response, error) in
+            
+            let httpResponse = response as? HTTPURLResponse
+            
+            os_log("Request for %s completed with response code: %s and headers %s",
+                   log: generalLog,
+                   type: .debug,
+                   url.absoluteString,
+                   String(describing: httpResponse?.statusCode),
+                   String(describing: httpResponse?.allHeaderFields))
+            
+            if let error = error {
+                completion(nil, error)
+            } else if let error = httpResponse?.validateStatusCode(statusCode: 200..<300) { 
```

Comment:
An alternative to this, we could potentially do the status code validation on the appropriate URLSessionDelegate callback, but after discussions with Brindy we decided it's unnecessary.
I made this default to returning the dataTask callback error over the status code error, as I figured it was likely to be more useful than a generic HTTPURLResponseError.invalidStatusCode error, but don't have strong options about it. (I think at any rate it's unlikely that they'll be no error at this point but an invalid status code)